Modules/_stat.c

```diff
@@ -492,113 +492,261 @@ ST_CTIME\n\
 ");
 
 
+static int
+stat_exec(PyObject *module)
+{
+    if (PyModule_AddIntMacro(module, S_IFDIR) < 0) {
+        return -1;
+    }
```

Comment:
This code looks to be basically copy-paste dozens of times. Maybe it would be worth it to add a macro to have a single single per added constant?

Reply:
Yeah that might be better